### PR TITLE
Feature/implement validator

### PR DESCRIPTION
**Pull request by nasdan**

### CODE REVIEW

File: examples/js/readme.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# is-number-validator example
+
```
Comment (brauliodiez) [inline examples/js/readme.md:2]: I will add a proposal text that we can reuse in other libraries

File: src/validator.spec.ts [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+    // Assert
+    expect(result).toEqual({
+      succeeded: false,
+      message: 'Must be a number',
+      type: 'IS_NUMBER',
+    });
+  });
+
+  it('should return failed validation when it feeds value equals null', () => {
```
Comment (brauliodiez) [inline src/validator.spec.ts]: I think for null, undefined or blank string validation should pass, if you want to control that add a required validator on the beginning of the list.

Why this behavior? I add a non mandatory field that has to be a number.

**Comment by brauliodiez** [comment]
Cannot find main readme.md?

File: src/validator.spec.ts (hunk 2 above)
Comment (nasdan) [reply src/validator.spec.ts]: You are right. I will change it

**Comment by nasdan** [comment]
About the main readme, It is on master branch because I've push project structure before this changes to get a cleaner PR. We can comment some changes if needed and i will apply it

File: src/validator.spec.ts (hunk 2 above)
Comment (nasdan) [reply src/validator.spec.ts]: Done

**Comment by brauliodiez** [comment]
Readme proposal:

# fonk-is-number-validator

This is a [fonk](https://github.com/Lemoncode/fonk) microlibrary that brings validation capabilities to:

- Validate if a field of a for is a number.

How to add it to an existing form validation schema:

We have the following form model:

```
const myFormValues = {
  product : 'shoes',
  price: 20,
}
```

We can add a isNumber validation to the myFormValues

```javascript
import { isNumber } from "@lemoncode/is-number-validator";

const validationSchema = {
  price: [isNumber.validator]
};
```

You can customize the error message displayed in two ways:

- Globally, replace the default error message in all validationSchemas (e.g. porting to spanish):

```javascript
import { setErrorMessage } from "@lemoncode/fonk/is-number-validator";

setErrorMessage("El campo debe de ser numérico");
```

- Locally just override the error message for this validationSchema:

```javascript
import { isNumber } from "@lemoncode/fonk/is-number-validator";

const validationSchema = {
  price: [
    {
      validator: isNumber.validator,
      message: "Error message only updated for the validation schema"
    }
  ]
};
```

Please, refer to [fonk](https://github.com/Lemoncode/fonk) to know more.

## License

[MIT](./LICENSE)

# About Basefactor + Lemoncode

We are an innovating team of Javascript experts, passionate about turning your ideas into robust products.

[Basefactor, consultancy by Lemoncode](http://www.basefactor.com) provides consultancy and coaching services.

[Lemoncode](http://lemoncode.net/services/en/#en-home) provides training services.

For the LATAM/Spanish audience we are running an Online Front End Master degree, more info: http://lemoncode.net/master-frontend

**Pull request merged**